### ISSUE TITLE
permit items to be have no visible form in bibliography

### ISSUE THREAD

**Issue by bwiernik** [citation-style-language/documentation#74]
Currently, it's assumed that all cited items will have a visible form in both `citation` and `bibliography` contexts (if the style has a bibliography). Some styles call for some items to only appear in citations, but not the bibliography (e.g., Vancouver and APA want  personal communications to appear with a specific formatting in text but be omitted from the bibliography). This could be handled by adding a `choose` element to show/hide some items conditionally (cf. [here](https://github.com/citation-style-language/styles/pull/3005)). 

citeprocj-js incidentally handles this gracefully, but the spec should make this an explicit expectation to ensure consistent behavior

**Comment by denismaier** [comment]
Citeproc-js actually has an option to exclude certain item types from the bibliography. If we adopt that solution, we will be able to handle these requirements without `choose` elements.

Anyway, I think this is a good suggestion, and also in line with the plans to implement multiple (filtered) bibliographies.

**Comment by bwiernik** [comment]
For APA rules at least, it’s more nuanced than certain item types. So `choose` will still be the easiest approach

**Comment by bdarcus** [comment]
I think this is part of the multi-section bib issue. Basically, you need a
section filtered on type that does not print.

**Comment by bdarcus** [comment]
> For APA rules at least, it’s more nuanced than certain item types. So `choose` will still be the easiest approach

Can you explain a bit?

**Comment by bwiernik** [comment]
A personal communication or interview that is retrievable (has a publisher, URL, or DOI) is included in the bibliography. A personal communication or interview that is not retrievable is only shown in text.

**Comment by bwiernik** [comment]
I think wrapping that case up in the multi-section bibliographies is way overkill and puts burden on the user that should be automated. 

Rules similar to APA are common. For example, many physical science journals require that unpublished data only appear in text, not in the bibliography. These styles currently include all items because the spec doesn’t allow for an item to have no printed form in the bibliography.

**Comment by bwiernik** [comment]
The user should not be required to know these simple automatable rules and manually implement them, especially when they can be implemented with just a minor change in instructions to processors.

The multi section bib framework is for manually-constructed sections that require user input, like primary and secondary sources.

**Comment by bdarcus** [comment]
Got it.

I have the hunch these may be at least complementary, and we should design
it to be so.

So what's the specific proposal; that all element content of choose should
be optional?

**Comment by bdarcus** [comment]
Multi-bib has different use cases, some of them automated.

**Comment by bwiernik** [comment]
No, just that “items must have a printed form in citations, but may or may not have a printed form in bibliographies”.

**Comment by bdarcus** [comment]
That's language for the spec?

I had checked, and the schema already does allow an empty if element, which
could be used to catch items to not print.

In any case, seems straightforward.

PR time?
